### ISSUE TITLE
`Cannot read property 'rows' of undefined` when aliasing column to whitespace

### ISSUE THREAD

**Issue by stearnsc** [metabase/metabase#14923]
**tldr**
Executing `SELECT 1 as " ";` in 0.38.0 returns a javascript error `Cannot read property 'rows' of undefined`.

Hi, we recently did a major upgrade to 0.38. One of our pulses failed to execute after the upgrade with the error `Cannot read property 'rows' of undefined`. After digging in, we found the culprit was a column we had aliased to `" "` for display purposes.

**Logs**
Browser
```
[re-reselect] Invalid cache key "undefined" has been returned by keySelector function.
error TypeError: Cannot read property 'rows' of undefined
app-main.bundle.js?fc344f13f5994a3b4142:5 Error getting setting table.columns TypeError: Cannot read property 'cols' of undefined
```
Server
```
[d9c7dbc4-f602-4bee-906e-eee7cc0fe0fb] 2021-02-22T07:51:01-07:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: null
{:database_id 2,
 :started_at #t "2021-02-22T14:51:01.481184Z[UTC]",
 :via
 [{:status :failed,
   :class clojure.lang.ExceptionInfo,
   :error "Error reducing result rows",
   :stacktrace
   ["--> query_processor.context.default$default_reducef$fn__36877.invoke(default.clj:60)"
    "query_processor.context.default$default_reducef.invokeStatic(default.clj:57)"
    "query_processor.context.default$default_reducef.invoke(default.clj:48)"
    "query_processor.context$reducef.invokeStatic(context.clj:69)"
    "query_processor.context$reducef.invoke(context.clj:62)"
    "query_processor.context.default$default_runf$respond_STAR___36881.invoke(default.clj:69)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:392)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:374)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:383)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:374)"
    "driver.sql_jdbc$fn__78967.invokeStatic(sql_jdbc.clj:54)"
    "driver.sql_jdbc$fn__78967.invoke(sql_jdbc.clj:52)"
    "query_processor.context$executef.invokeStatic(context.clj:59)"
    "query_processor.context$executef.invoke(context.clj:48)"
    "query_processor.context.default$default_runf.invokeStatic(default.clj:68)"
    "query_processor.context.default$default_runf.invoke(default.clj:66)"
    "query_processor.context$runf.invokeStatic(context.clj:45)"
    "query_processor.context$runf.invoke(context.clj:39)"
    "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
    "query_processor.reducible$pivot.invoke(reducible.clj:31)"
    "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__46313.invoke(mbql_to_native.clj:25)"
    "query_processor.middleware.check_features$check_features$fn__45589.invoke(check_features.clj:41)"
    "query_processor.middleware.limit$limit$fn__46299.invoke(limit.clj:37)"
    "query_processor.middleware.cache$maybe_return_cached_results$fn__45240.invoke(cache.clj:211)"
    "query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__46478.invoke(optimize_datetime_filters.clj:133)"
    "query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__44386.invoke(auto_parse_filter_values.clj:43)"
    "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__40767.invoke(wrap_value_literals.clj:147)"
    "query_processor.middleware.annotate$add_column_info$fn__40630.invoke(annotate.clj:582)"
    "query_processor.middleware.permissions$check_query_permissions$fn__45464.invoke(permissions.clj:69)"
    "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__47001.invoke(pre_alias_aggregations.clj:40)"
    "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__45662.invoke(cumulative_aggregations.clj:60)"
    "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__47314.invoke(resolve_joined_fields.clj:94)"
    "query_processor.middleware.resolve_joins$resolve_joins$fn__47619.invoke(resolve_joins.clj:178)"
    "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__43976.invoke(add_implicit_joins.clj:181)"
    "query_processor.middleware.large_int_id$convert_id_to_string$fn__46274.invoke(large_int_id.clj:44)"
    "query_processor.middleware.format_rows$format_rows$fn__46254.invoke(format_rows.clj:74)"
    "query_processor.middleware.desugar$desugar$fn__45728.invoke(desugar.clj:21)"
    "query_processor.middleware.binning$update_binning_strategy$fn__44746.invoke(binning.clj:228)"
    "query_processor.middleware.resolve_fields$resolve_fields$fn__45264.invoke(resolve_fields.clj:24)"
    "query_processor.middleware.add_dimension_projections$add_remapping$fn__43606.invoke(add_dimension_projections.clj:314)"
    "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__43837.invoke(add_implicit_clauses.clj:146)"
    "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__48049.invoke(upgrade_field_literals.clj:45)"
    "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__44133.invoke(add_source_metadata.clj:124)"
    "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__47198.invoke(reconcile_breakout_and_order_by_bucketing.clj:97)"
    "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__44333.invoke(auto_bucket_datetimes.clj:139)"
    "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__45311.invoke(resolve_source_table.clj:45)"
    "query_processor.middleware.parameters$substitute_parameters$fn__46983.invoke(parameters.clj:111)"
    "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__45363.invoke(resolve_referenced.clj:79)"
    "query_processor.middleware.expand_macros$expand_macros$fn__45984.invoke(expand_macros.clj:155)"
    "query_processor.middleware.add_timezone_info$add_timezone_info$fn__44142.invoke(add_timezone_info.clj:15)"
    "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__47985.invoke(splice_params_in_response.clj:32)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__47209$fn__47213.invoke(resolve_database_and_driver.clj:31)"
    "driver$do_with_driver.invokeStatic(driver.clj:60)"
    "driver$do_with_driver.invoke(driver.clj:56)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__47209.invoke(resolve_database_and_driver.clj:25)"
    "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__46202.invoke(fetch_source_query.clj:264)"
    "query_processor.middleware.store$initialize_store$fn__47994$fn__47995.invoke(store.clj:11)"
    "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
    "query_processor.store$do_with_store.invoke(store.clj:38)"
    "query_processor.middleware.store$initialize_store$fn__47994.invoke(store.clj:10)"
    "query_processor.middleware.validate$validate_query$fn__48056.invoke(validate.clj:10)"
    "query_processor.middleware.normalize_query$normalize$fn__46326.invoke(normalize_query.clj:22)"
    "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__43994.invoke(add_rows_truncated.clj:35)"
    "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__47970.invoke(results_metadata.clj:147)"
    "query_processor.middleware.constraints$add_default_userland_constraints$fn__45605.invoke(constraints.clj:42)"
    "query_processor.middleware.process_userland_query$process_userland_query$fn__47072.invoke(process_userland_query.clj:135)"
    "query_processor.middleware.catch_exceptions$catch_exceptions$fn__45548.invoke(catch_exceptions.clj:173)"
    "query_processor.reducible$async_qp$qp_STAR___37408$thunk__37409.invoke(reducible.clj:103)"
    "query_processor.reducible$async_qp$qp_STAR___37408.invoke(reducible.clj:109)"
    "query_processor.reducible$sync_qp$qp_STAR___37417$fn__37420.invoke(reducible.clj:135)"
    "query_processor.reducible$sync_qp$qp_STAR___37417.invoke(reducible.clj:134)"
    "query_processor$process_userland_query.invokeStatic(query_processor.clj:237)"
    "query_processor$process_userland_query.doInvoke(query_processor.clj:233)"
    "query_processor$fn__48102$process_query_and_save_execution_BANG___48111$fn__48114.invoke(query_processor.clj:249)"
    "query_processor$fn__48102$process_query_and_save_execution_BANG___48111.invoke(query_processor.clj:241)"
    "query_processor$fn__48146$process_query_and_save_with_max_results_constraints_BANG___48155$fn__48158.invoke(query_processor.clj:261)"
    "query_processor$fn__48146$process_query_and_save_with_max_results_constraints_BANG___48155.invoke(query_processor.clj:254)"
    "api.dataset$fn__54256$fn__54259.invoke(dataset.clj:55)"
    "query_processor.streaming$streaming_response_STAR_$fn__54237$fn__54238.invoke(streaming.clj:72)"
    "query_processor.streaming$streaming_response_STAR_$fn__54237.invoke(streaming.clj:71)"
    "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
    "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
    "async.streaming_response$do_f_async$fn__16055.invoke(streaming_response.clj:84)"],
   :error_type :qp,
   :ex-data {:type :qp}}],
 :error_type :invalid-query,
 :json_query {:type "native", :native {:query "select 1 as \" \";", :template-tags {}}, :database 2, :parameters [], :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> query_processor.middleware.annotate$fn__40554$deduplicate_cols_names__40559.invoke(annotate.clj:503)"
  "query_processor.middleware.annotate$fn__40582$merged_column_info__40587$fn__40591.invoke(annotate.clj:550)"
  "query_processor.middleware.annotate$fn__40582$merged_column_info__40587.invoke(annotate.clj:543)"
  "query_processor.middleware.annotate$add_column_info_xform$combine__40623.invoke(annotate.clj:574)"
  "query_processor.reducible$combine_additional_reducing_fns$fn__37431.invoke(reducible.clj:206)"
  "query_processor.middleware.limit$limit_xform$fn__46296.invoke(limit.clj:21)"
  "query_processor.context.default$default_reducef$fn__36877.invoke(default.clj:58)"
  "query_processor.context.default$default_reducef.invokeStatic(default.clj:57)"
  "query_processor.context.default$default_reducef.invoke(default.clj:48)"
  "query_processor.context$reducef.invokeStatic(context.clj:69)"
  "query_processor.context$reducef.invoke(context.clj:62)"
  "query_processor.context.default$default_runf$respond_STAR___36881.invoke(default.clj:69)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:392)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:374)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:383)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:374)"
  "driver.sql_jdbc$fn__78967.invokeStatic(sql_jdbc.clj:54)"
  "driver.sql_jdbc$fn__78967.invoke(sql_jdbc.clj:52)"
  "query_processor.context$executef.invokeStatic(context.clj:59)"
  "query_processor.context$executef.invoke(context.clj:48)"
  "query_processor.context.default$default_runf.invokeStatic(default.clj:68)"
  "query_processor.context.default$default_runf.invoke(default.clj:66)"
  "query_processor.context$runf.invokeStatic(context.clj:45)"
  "query_processor.context$runf.invoke(context.clj:39)"
  "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
  "query_processor.reducible$pivot.invoke(reducible.clj:31)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__46313.invoke(mbql_to_native.clj:25)"
  "query_processor.middleware.check_features$check_features$fn__45589.invoke(check_features.clj:41)"
  "query_processor.middleware.limit$limit$fn__46299.invoke(limit.clj:37)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__45240.invoke(cache.clj:211)"
  "query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__46478.invoke(optimize_datetime_filters.clj:133)"
  "query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__44386.invoke(auto_parse_filter_values.clj:43)"
  "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__40767.invoke(wrap_value_literals.clj:147)"
  "query_processor.middleware.annotate$add_column_info$fn__40630.invoke(annotate.clj:582)"
  "query_processor.middleware.permissions$check_query_permissions$fn__45464.invoke(permissions.clj:69)"
  "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__47001.invoke(pre_alias_aggregations.clj:40)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__45662.invoke(cumulative_aggregations.clj:60)"
  "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__47314.invoke(resolve_joined_fields.clj:94)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__47619.invoke(resolve_joins.clj:178)"
  "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__43976.invoke(add_implicit_joins.clj:181)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__46274.invoke(large_int_id.clj:44)"
  "query_processor.middleware.format_rows$format_rows$fn__46254.invoke(format_rows.clj:74)"
  "query_processor.middleware.desugar$desugar$fn__45728.invoke(desugar.clj:21)"
  "query_processor.middleware.binning$update_binning_strategy$fn__44746.invoke(binning.clj:228)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__45264.invoke(resolve_fields.clj:24)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__43606.invoke(add_dimension_projections.clj:314)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__43837.invoke(add_implicit_clauses.clj:146)"
  "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__48049.invoke(upgrade_field_literals.clj:45)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__44133.invoke(add_source_metadata.clj:124)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__47198.invoke(reconcile_breakout_and_order_by_bucketing.clj:97)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__44333.invoke(auto_bucket_datetimes.clj:139)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__45311.invoke(resolve_source_table.clj:45)"
  "query_processor.middleware.parameters$substitute_parameters$fn__46983.invoke(parameters.clj:111)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__45363.invoke(resolve_referenced.clj:79)"
  "query_processor.middleware.expand_macros$expand_macros$fn__45984.invoke(expand_macros.clj:155)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__44142.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__47985.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__47209$fn__47213.invoke(resolve_database_and_driver.clj:31)"
  "driver$do_with_driver.invokeStatic(driver.clj:60)"
  "driver$do_with_driver.invoke(driver.clj:56)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__47209.invoke(resolve_database_and_driver.clj:25)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__46202.invoke(fetch_source_query.clj:264)"
  "query_processor.middleware.store$initialize_store$fn__47994$fn__47995.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
  "query_processor.store$do_with_store.invoke(store.clj:38)"
  "query_processor.middleware.store$initialize_store$fn__47994.invoke(store.clj:10)"
  "query_processor.middleware.validate$validate_query$fn__48056.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__46326.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__43994.invoke(add_rows_truncated.clj:35)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__47970.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__45605.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__47072.invoke(process_userland_query.clj:135)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__45548.invoke(catch_exceptions.clj:173)"
  "query_processor.reducible$async_qp$qp_STAR___37408$thunk__37409.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___37408.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___37417$fn__37420.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___37417.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:237)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:233)"
  "query_processor$fn__48102$process_query_and_save_execution_BANG___48111$fn__48114.invoke(query_processor.clj:249)"
  "query_processor$fn__48102$process_query_and_save_execution_BANG___48111.invoke(query_processor.clj:241)"
  "query_processor$fn__48146$process_query_and_save_with_max_results_constraints_BANG___48155$fn__48158.invoke(query_processor.clj:261)"
  "query_processor$fn__48146$process_query_and_save_with_max_results_constraints_BANG___48155.invoke(query_processor.clj:254)"
  "api.dataset$fn__54256$fn__54259.invoke(dataset.clj:55)"
  "query_processor.streaming$streaming_response_STAR_$fn__54237$fn__54238.invoke(streaming.clj:72)"
  "query_processor.streaming$streaming_response_STAR_$fn__54237.invoke(streaming.clj:71)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
  "async.streaming_response$do_f_async$fn__16055.invoke(streaming_response.clj:84)"],
 :context :ad-hoc,
 :error
 "Input to deduplicate-cols-names does not match schema: \n\n\t   [(named [{:field_ref (named [nil (named (not (\"Non-blank string\" \" \")) \"field-name\") nil] \"Must be a valid instance of one of these clauses: :field-id, :field-literal, :joined-field, :fk->, :datetime-field, :expression, :binning-strategy\")}] cols)]  \n\n",
 :row_count 0,
 :running_time 0,
 :ex-data
 {:type :schema.core/error,
  :value [[{:display_name " ", :source :native, :field_ref [:field-literal " " :type/Integer], :name " ", :base_type :type/Integer}]],
  :error
  [(named [{:field_ref (named [nil (named (not ("Non-blank string" " ")) "field-name") nil] "Must be a valid instance of one of these clauses: :field-id, :field-literal, :joined-field, :fk->, :datetime-field, :expression, :binning-strategy")}] cols)]},
 :data {:rows [], :cols []}}
```

**To Reproduce**
In the "Ask a question" section, asking a native SQL question against postgres, ask
```postgresql
SELECT 1 as " ";
```
you should get an error - `Cannot read property 'rows' of undefined`.

**Expected behavior**
Either an error telling us that whitespace-only aliases are unsupported, or a successful query with the whitespace-aliased column name.

**Information about your Metabase Installation:**
Version v0.38.0
Built on 2021-02-16
Running in Docker
Our db - Postgres
Metabase internal db - Postgres

**Severity**
Not severe at all, once we identified the problem. Was somewhat tricky to debug, because it happened deep in a complicated query, but now that we know the restriction, it's not really a problem at all.

**Comment by flamber** [comment]
I cannot reproduce on 0.41.2, but there were a lot of query processor changes in the past couple of major versions.
